infrastructure/datastore/csv.go

```diff
@@ -6,34 +6,90 @@ import (
 	"os"
 )
 
+/*
+	Wrapper of the CSV file, combines the management
+	of the os and enconding/csv libraries.
+*/
 type CSVHandler struct {
-	Path   string
-	File   *os.File
-	Writer *csv.Writer
-	Reader *csv.Reader
+	path      string
+	schema    []string
+	readFile  *os.File
+	writeFile *os.File
+	writer    *csv.Writer
+	reader    *csv.Reader
 }
 
+/*
+	Returns an instance of the CSVHandler
+*/
 func NewCSVHandler(filePath string) *CSVHandler {
-	return &CSVHandler{Path: filePath}
+	return &CSVHandler{path: filePath}
 }
 
+/*
+	Method used to assign the handler with the required
+	tools to manage the csv file, is separated of the
+	initialization of the handler in order to avoid leaving
+	the CSV file opened in case something fails.
+*/
 func (h *CSVHandler) BuildHandler() error {
-	f, err := os.Open(h.Path)
+	wf, err := os.OpenFile(h.path, os.O_WRONLY|os.O_APPEND, 0666)
+
+	if err != nil {
+		return errors.New("CANNOT OPEN CSV FILE")
```

Comment:
Do not dynamically create errors, create them in the errors package and use them here. `Sentence case` and wrap errors before returning them.

Fix all other occurrences.